### PR TITLE
 Allow adapter to throw explicit error for notifications

### PR DESCRIPTION
**Pull request by mikaelkalt**
Added the possibility that the adapter can throw an exception in the case that the notification could not be processed and the match should be notified again. 

On the server side I'm now handling the case that a notification could not be sent - in this case it is not stored as already notified, and would be notified the next time the node is available. 

Therefore I needed to do a bigger refactoring for the whole matching storage - but this way it might covers already the case of having multiple commencements for the same termination.

### CODE REVIEW

**Comment by mikaelkalt** [comment]
Please do not yet merge. I want to discuss that first.

File: hub/src/main/java/ch/prevo/open/hub/nodes/NodeService.java [hunk 1]
```diff
@@ -51,23 +48,21 @@ public NodeService(NodeRegistry nodeRegistry, MatcherService matcherService, Nod
...
             List<InsurantInformation> filteredInformation = filterInvalidAndAlreadyMatchedEntries(nodeConfig,
-                    pensionFundEntries,
-                    matcherService::employmentCommencementNotMatched);
+                    pensionFundEntries);
             entries.addAll(filteredInformation);
         }
         return entries;
     }
 
     private List<InsurantInformation> filterInvalidAndAlreadyMatchedEntries(NodeConfiguration nodeConfiguration,
```
Comment (lennartx) [inline hub/src/main/java/ch/prevo/open/hub/nodes/NodeService.java]: Rename to reflect the change. E:g. `filterInvalidEntries`

File: hub/src/main/java/ch/prevo/open/hub/nodes/NodeCaller.java [hunk 2]
```diff
@@ -1,21 +1,22 @@
...
-import static java.util.Arrays.asList;
-import static java.util.Collections.emptyList;
+import ch.prevo.open.encrypted.model.CapitalTransferInformation;
+import ch.prevo.open.encrypted.model.CommencementMatchNotification;
+import ch.prevo.open.encrypted.model.InsurantInformation;
+import ch.prevo.open.encrypted.model.TerminationMatchNotification;
+import ch.prevo.open.hub.repository.NotificationRepository;
 
 @Service
 public class NodeCaller {
```
Comment (lennartx) [inline hub/src/main/java/ch/prevo/open/hub/nodes/NodeCaller.java]: Would try to keep this class as free of non-rest logic as possible, and rather do the NotificationRepository calls in the NodeService

Comment (mikaelkalt) [reply hub/src/main/java/ch/prevo/open/hub/nodes/NodeCaller.java]: What's the advantage of it? The idea behind it, was that the caller doesn't have to be aware of already notified matches. E.g. if someone else would use the `NodeCaller#postCommencementNotification` one needs to check again if this notification was not already sent?

**Pull request merged**